ui/components/failed.jsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import ImageDiff from 'react-image-diff';
+import {Clearfix, ButtonGroup, Button,
+  Jumbotron, Grid, Row, Col, Panel}
+  from 'react-bootstrap';
+
+function _renderDefaultView(diff) {
+  return <div className="simple">
+      <img className="diff"
+           src={diff}
+           key={diff}
+      />
+    </div>;
+}
+
+function _render2UpView(baseline, screenshot) {
+  return <div className="simple">
+      <Grid>
+        <Row className="show-grid">
+          <Col xs={12} md={6}>
+            <img className="baseline"
+                 src={baseline}
+                 key={baseline} />
+          </Col>
+          <Col xs={12} md={6}>
+            <img className="screenshot"
+                 src={screenshot}
+                 key={screenshot} />
+          </Col>
+        </Row>
+    </Grid>
+  </div>;
+}
+
+function _renderSpecialView(baseline, screenshot, view, value, changeValue) {
+  return <div className="special">
+      <ImageDiff before={screenshot}
+                 after={baseline}
+                 type={view}
+                 value={value}
+      />
+      <Clearfix/>
+      <input type="range"
+             min={0}
+             max={1}
+             step={.01}
+             defaultValue={value}
+             onChange={changeValue}
+      />
+    </div>;
+}
+
+module.exports = React.createClass({
```

Comment:
Can you add an empty line between each key of this object? You can take a look in how we write components in web-ui. It makes things easier to read ❤️ .